src/controllers/user/Controller.ts

```
@@ -0,0 +1,86 @@
+                    message: 'User update  successfully',
+                    status: 200,
+                });
+            }
+        } catch (e) {
+            console.error(e);
+        }
+    }
+    public async deleteUser(req, res, next) {
+        try {
+            const deleteUser = await userRepository.delete({_id: req.params.id});
+            if (deleteUser === ' not found in delete ') {
+                next({
+                    message: deleteUser,
+                    status: 404,
+                });
+            }
+            else {
+                res.send({
+                data: req.params.id,
+                message: 'User delete   successfully',
+                status: 200,
+                });
+            }
+        }
+        catch (e) {
+            console.error(e);
+        }
+    }
+    public getUser( req, res) {
+        res.send({
+            message: 'Me',
+            status: 'ok',
+            data: req.user,
+        });
+    }
+    public async login(req, res, next) {
+        // console.log('inside login request ::::', req.body);
+        try {
+            const{ email, password } = req.body;
+            const userlogin = await userRepository.findOne({ email });
+            // console.log('User is :::::', userlogin);
+            if (!userlogin) {
+                return next('User Not found');
+            }
+            const {  password : hashPassword } = userlogin;
+            if (!(bcrypt.compareSync(password , hashPassword))) {
+                return next('password does not match');
+            }
+            const token = jwt.sign(userlogin, config.secretKey, {expiresIn: '15m'});
+            // console.log('Token is ::::', token);
+            // console.log('User Response', user);
+            res.send({
+                message: 'Authorization Token',
+                status: 200,
+                data: token,
+            });
+        }
+        catch (e) {
+            console.error(e);
```

here as well, We should return a proper error.